protocol/check.go

```diff
@@ -0,0 +1,33 @@
+package protocol
+
+import (
+	"bytes"
+	"context"
+	"errors"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
```

Comment:
Can you document:
* Why we need this constant. Why the similar value in the generated swap factory code can not be used.
* How to update this constant the next time the contract code changes.
* How the offsets in `CheckContractCode` were derived and how they work.

Reply:
I did some research and I guess the problem is [immutable variables](https://docs.sourcify.dev/docs/immutables/).

Reply:
updated, let me know if the comments make sense